### PR TITLE
Image Component - Primitive Componets - [WSTEAMA-144]

### PR DESCRIPTION
**Pull request by pvaliani**
Resolves JIRA ISSUE WSTEAMA-144

**Overall change:**
- Creates a new `Image` primitive component
- The `Image` component is used to render an image using the correct custom style for for the service. 
- The `Image` component supports pre-load and lazy loading and can render a placeholder when image is loading. 
- The `Image` component supports webP in the first instance and makes use of jpeg as a fallback format.
- The `Image` component is supported on AMP and Canonical pages.
- [README](https://github.com/bbc/simorgh/blob/194ee84f2bc0d5b4cd125c38368c1d28a882aafe/src/app/components/Image/README.md)

**Code changes:**

- Implements the `Image` component using emotion theming.
- Adds unit tests for Canonical and AMP.
- Adds a story page for Storybook, co-located styles and co-located placeholder image.

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Comment by greenc05** [comment]
Would this PR be the right place to make an improvement to reduce the displayed image size when there is no CSS?

File: src/app/components/Image/index.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  return (
+    <React.Fragment>
+      {preload && (
+        <Helmet>
+          <link
+            rel="preload"
+            as="image"
+            href={src}
+            imagesrcset={primarySrcset}
+            imagesizes={sizes}
```
Comment (andrewscfc) [inline src/app/components/Image/index.tsx:74]: Any idea why these checks are failing are these non-standard attributes?

File: src/app/components/Image/index.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            paddingBottom: `${(1 / wrapperAspectRatio) * 100}%`,
+          },
+        ]}
+      >
+        <div
+          css={[placeholder && styles.placeholder, styles.placeholderWrapper]}
+        >
+          {isAmp ? (
+            <amp-img
+              alt={alt}
```
Comment (andrewscfc) [inline src/app/components/Image/index.tsx]: good old amp, are there proper types for amp attributes?

**Review by andrewscfc — APPROVED** [review]
LGTM, few type issues, unsure how much they matter?

File: src/app/components/Image/index.tsx (hunk 2 above)
Comment (pvaliani) [reply src/app/components/Image/index.tsx]: There aren't typescript types for AMP but we're looking at a few workarounds so PR isn't ready for full review quite just yet

File: src/app/components/Image/index.tsx (hunk 1 above)
Comment (jroebu14) [reply src/app/components/Image/index.tsx:74]: they seem to be missing from these attributes from the link type. I've added them in [here](https://github.com/bbc/simorgh/pull/10265/files#diff-e044a0734938281ee2184aeb22e60ae3502717d53b8f522328ac7ec4685a459bR13-R14).

File: src/app/components/Image/index.tsx (hunk 2 above)
Comment (jroebu14) [reply src/app/components/Image/index.tsx]: Following the advice [here](https://nextjs.org/docs/advanced-features/amp-support/typescript) I've addressed this [here](https://github.com/bbc/simorgh/pull/10265/files#diff-e044a0734938281ee2184aeb22e60ae3502717d53b8f522328ac7ec4685a459b)

**Comment by jroebu14** [comment]
> Would this PR be the right place to make an improvement to reduce the displayed image size when there is no CSS?

@greenc05 the displayed image size when there is no CSS is controlled by the `width` and `height` properties on the `img` element. These properties come from the fetched page data. I'm not sure why they are so big in some cases. But I think the fix would need to be in the page data API.

Screenshot 2022-09-08 at 10 09 02
Screenshot 2022-09-08 at 10 05 58

File: src/app/components/Image/index.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  isAmp = false,
+  lazyLoad = false,
+  placeholder = true,
+  preload = false,
+  primaryMediaType,
+  primarySrcset,
+  sizes = '100vw',
+  src,
+  width,
+}: React.PropsWithChildren<Props>) => {
```
Comment (amoore108) [inline src/app/components/Image/index.tsx]: Minor thing but you could import this as `{PropsWithChildren} from react`, similar case below with `Fragment`.

**Review by amoore108 — APPROVED** [review]
Looks great! 👍🏼

File: src/app/components/Image/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Image
+
+## Description
+
+The `Image` component is used to render an image using the correct custom style for for the service. It supports pre-load and lazy loading and can render a placeholder when image is loading. The image component supports webP in the first instance and makes use of jpeg as a fallback format. The image component supports AMP and canonical pages.
```
Comment (greenc05) [inline src/app/components/Image/README.md]: Typo '...custom style for for the service...' one too many 'for'

File: src/app/components/Image/index.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            imagesrcset={srcSet}
+            imagesizes={sizes}
+          />
+        </Helmet>
+      )}
+      <div
+        className={className}
+        css={[styles.wrapper, placeholder && styles.placeholder]}
+        style={{
+          paddingBottom: `${roundNumber((1 / wrapperAspectRatio) * 100)}%`,
```
Comment (andrewscfc) [inline src/app/components/Image/index.tsx:82]: Just a question, why does the `paddingBottom` get added based on the aspect ratio?

**Review by andrewscfc — APPROVED** [review]

File: src/app/components/Image/index.tsx (hunk 5 above)
Comment (jroebu14) [reply src/app/components/Image/index.tsx:82]: it's an age-old css hack https://nikitahl.com/css-aspect-ratio

**Pull request merged**